test/src/responder/multiResponder.test.ts

```diff
@@ -0,0 +1,277 @@
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+
+        await responder.startResponse(appointmentId, responseData);
+        const item = responder.queue.queueItems[0];
+
+        await responder.startResponse(appointmentId2, responseData2);
+        const itemAfterReplace = responder.queue.queueItems[0];
+
+        await responder.txMined(item.request.identifier, item.nonce);
+        const itemAfterMined = responder.queue.queueItems[0];
+
+        expect(responder.queue.queueItems.length).to.equal(1);
+        expect(itemAfterMined.request.identifier).to.deep.equal(itemAfterReplace.request.identifier);
+        expect(itemAfterMined.nonce).to.equal(itemAfterReplace.nonce + 1);
+        verify(transactionTrackerMock.addTx(anything(), anything())).times(4);
+    });
+
+    it("txMined does nothing when queue is empty", async () => {
+        const responder = new MultiResponder(
+            signer,
+            increasingGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+
+        await responder.txMined(new PisaTransactionIdentifier(1, "data", "to", new BigNumber(0), new BigNumber(10)), 1);
+        expect(responder.queue.queueItems.length).to.equal(0);
+        verify(transactionTrackerMock.addTx(anything(), anything())).never();
+    });
+
+    it("txMined does nothing when item not in queue", async () => {
+        const appointmentId = "app1";
+        const responseData = createResponseData("app1");
+        const responder = new MultiResponder(
+            signer,
+            increasingGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+        await responder.startResponse(appointmentId, responseData);
+        const queueBefore = responder.queue;
+        await responder.txMined(new PisaTransactionIdentifier(1, "data", "to", new BigNumber(0), new BigNumber(10)), 1);
+
+        expect(responder.queue).to.equal(queueBefore);
+        verify(transactionTrackerMock.addTx(anything(), anything())).once();
+    });
+
+    it("txMined does nothing nonce is not front of queue", async () => {
+        const appointmentId = "app1";
+        const responseData = createResponseData("app1");
+        const responder = new MultiResponder(
+            signer,
+            increasingGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+        await responder.startResponse(appointmentId, responseData);
+        const queueBefore = responder.queue;
+        const item = responder.queue.queueItems[0];
+        await responder.txMined(item.request.identifier, item.nonce + 1);
+
+        expect(responder.queue).to.equal(queueBefore);
+        verify(transactionTrackerMock.addTx(anything(), anything())).once();
+    });
+});
```

Comment:
Perhaps add one or two tests for TransactionTracker.

Reply:
My plan is to remove the transaction tracker completley and replace it with an anchor state reducer